src/photo.go

```diff
@@ -0,0 +1,171 @@
+package main
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+
+	exiftool "github.com/barasher/go-exiftool"
+)
+
+// Photo , structure to store pictures information
+type Photo struct {
+	photoPath    string
+	labels       []string
+	gps          string // TODO review this type
+	dateCreation time.Time
+	fileType     string
+	imageSize    string
+	device       string
+	metadata     exiftool.FileMetadata // Raw data extracted from the file
+}
+
+// NewPhoto , return Photo struct with nil for the metadata
+func NewPhoto(filePath string) *Photo {
+	return &Photo{
+		photoPath: filePath,
+	}
+}
+
+// LoadPhotoTags , Load metadata informations from the file on the harddrive
+func (p *Photo) LoadPhotoTags() error {
+
+	// Initialize exiftool
+	et, err := exiftool.NewExiftool()
+	if err != nil {
+		return err
+	}
+	defer et.Close()
+
+	// Extract Metadata, we can provide a list of photo
+	fileInfos := et.ExtractMetadata(p.photoPath)
+	if len(fileInfos) == 1 {
+		//Extract the metadata information for the first file
+		p.metadata = fileInfos[0]
+	} else {
+		return fmt.Errorf("Unable to load exif metadata for :%s", p.photoPath)
+	}
+
+	// return error from the exiftool for the pressed file
+	if p.metadata.Err != nil {
+		return p.metadata.Err
+	}
+
+	return nil
+
+}
+
+// SetPhotoStruct , get information from the raw data and set fields in the struct
+func (p *Photo) SetPhotoStruct() error {
+
+	var err error
+	// TODO change to use the multiple error mechanism
```

Comment:
This must be changed to use : go-multierror 
I will come back later, I would like to add more value before.